janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java

```diff
@@ -419,7 +421,7 @@ public Boolean get(byte[] array, int offset, int limit) {
             pos++;
         }
         assert offset<=data.length;
-        if (data.length>offset*3/2) {
+        if (data.length/3>offset/2) {
```

Comment:
Good idea, let me investigate that.

Reply:
FP division is sometimes faster than integer division so using floating point here might actually be faster. Anyways, a division is usually much slower than multiplication, thus I changed it to the multiplication form.

In the old version, integer overflow happens as long as offset > (Integer.MAX_VALUE / 3) and shrink will happen, while in the new version, integer overflow happens when offset * 1.5 is larger than Integer.MAX_VALUE.

There is no much difference to be honest since whether to shrink does not make too much difference.

Reply:
> In the old version, integer overflow happens as long as offset * 3 is larger than Integer.MAX_VALUE and shrink will happen, while in the new version, integer overflow happens when offset * 1.5 is larger than Integer.MAX_VALUE.

Constants are usually calculated during the compilation (that said, not sure about division operator but I guess it's computed as well). I didn't check but I guess compiler will compute `3/2` during the compilation. Thus the old form would look like `data.length>offset*1.5`. Thus, I guess the logic didn't change (again, not sure because didn't check). Instead of `data.length>offset*1.5` we now use `data.length > offset + (offset >> 1)`. I didn't check the performance between these approaches but I guess your approach is faster. 
Again, we can invert division to be multiplication and back. For example, instead of `(data.length/1.5f>offset)` I could write `(data.length*(2/3)>offset)` which is essentially `(data.length*0.66666...>offset)`. 
I don't know how Java computes divisions and multiplications with float or double values and don't know the performance differences. I know that generally addition and subtraction is faster then multiplication and multiplication is faster then division but I don't know how much to be fair. 
I personally good with any solution you use here.

Reply:
Not really. See the sample below:
```java
public class IntegerOverflowApp {
    public static void main(String[] args) {
        int veryLargeNum = Integer.MAX_VALUE / 2;
        System.out.println(veryLargeNum * 3 / 2); // result is: -536870913
        System.out.println(veryLargeNum * 1.5); // result is: 1.6106127345E9
        System.out.println(veryLargeNum + veryLargeNum >> 1); // result is: 1073741823
    }
}
```
The compiler cannot optimize 3/2 because it would change the overflow behavior (let alone floating point error - the second and the third result are quite different). Not sure whether JIT will do this optimization but I don't think it can due to the same reason.

Reply:
Totally makes sense. Thank you for checking that!